test/functional/wallet_migration.py

```diff
@@ -0,0 +1,405 @@
+
+        pub1 = multisig0.getaddressinfo(addr1)["pubkey"]
+        pub2 = multisig0.getaddressinfo(addr2)["pubkey"]
+
+        # Some keys in multisig do not belong to this wallet
+        self.log.info("Test migration of a wallet that has some keys in a multisig")
+        self.nodes[0].createwallet(wallet_name="multisig1")
+        multisig1 = self.nodes[0].get_wallet_rpc("multisig1")
+        ms_info = multisig1.addmultisigaddress(2, [multisig1.getnewaddress(), pub1, pub2])
+        ms_info2 = multisig1.addmultisigaddress(2, [multisig1.getnewaddress(), pub1, pub2])
+        assert_equal(multisig1.getwalletinfo()["descriptors"], False)
+
+        addr = ms_info["address"]
+        txid = default.sendtoaddress(addr, 10)
+        multisig1.importaddress(addr)
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["ismine"], False)
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["iswatchonly"], True)
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["solvable"], True)
+        self.generate(self.nodes[0], 1)
+        multisig1.gettransaction(txid)
+        assert_equal(multisig1.getbalances()["watchonly"]["trusted"], 10)
+        assert_equal(multisig1.getaddressinfo(ms_info2["address"])["ismine"], False)
+        assert_equal(multisig1.getaddressinfo(ms_info2["address"])["iswatchonly"], False)
+        assert_equal(multisig1.getaddressinfo(ms_info2["address"])["solvable"], True)
+
+        # Migrating multisig1 should see the multisig is no longer part of multisig1
+        # A new wallet multisig1_watchonly is created which has the multisig address
+        # Transaction to multisig is in multisig1_watchonly and not multisig1
+        multisig1.migratewallet()
+        assert_equal(multisig1.getwalletinfo()["descriptors"], True)
+        self.assert_is_sqlite("multisig1")
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["ismine"], False)
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["iswatchonly"], False)
+        assert_equal(multisig1.getaddressinfo(ms_info["address"])["solvable"], False)
+        assert_raises_rpc_error(-5, "Invalid or non-wallet transaction id", multisig1.gettransaction, txid)
+        assert_equal(multisig1.getbalance(), 0)
+        assert_equal(multisig1.listtransactions(), [])
+
+        assert_equal("multisig1_watchonly" in self.nodes[0].listwallets(), True)
+        ms1_watchonly = self.nodes[0].get_wallet_rpc("multisig1_watchonly")
+        ms1_wallet_info = ms1_watchonly.getwalletinfo()
+        assert_equal(ms1_wallet_info['descriptors'], True)
+        assert_equal(ms1_wallet_info['private_keys_enabled'], False)
+        self.assert_is_sqlite("multisig1_watchonly")
+        assert_equal(ms1_watchonly.getaddressinfo(ms_info["address"])["ismine"], True)
+        assert_equal(ms1_watchonly.getaddressinfo(ms_info["address"])["solvable"], True)
+        assert_equal(ms1_watchonly.getaddressinfo(ms_info2["address"])["ismine"], False)
```

Comment:
fb5e0957e066bd0825947a399242242f1f1a2ae6: maybe comment here that, because it wasn't watch-only, it doesn't get moved to the watch-only wallet, but to the solvables wallet (as shown below in the test)

Reply:
Done